router-tests/safelist_test.go

```diff
@@ -0,0 +1,165 @@
+package integration
+
+import (
+	"github.com/stretchr/testify/require"
+	"github.com/wundergraph/cosmo/router-tests/testenv"
+	"github.com/wundergraph/cosmo/router/core"
+	"github.com/wundergraph/cosmo/router/pkg/config"
+	"go.uber.org/zap/zapcore"
+	"net/http"
+	"testing"
+)
+
+func TestSafelist(t *testing.T) {
+	t.Parallel()
+
+	var (
+		persistedQuery        = "query Employees {\n employees {\n id\n }\n}"
+		nonPersistedQuery     = "query Employees {\n\n\n employees {\n id\n }\n}"
+		queryWithDetails      = "query Employees {\n employees {\n id\n details {\n forename\n} \n}\n}"
+		persistedNotFoundResp = `{"errors":[{"message":"PersistedQueryNotFound","extensions":{"code":"PERSISTED_QUERY_NOT_FOUND"}}]}`
+	)
+
```

Comment:
We should also test if it is possible to run a regular persisted operation when safelist is enabled.